pkg/data/data_store.go

```diff
@@ -40,19 +41,38 @@ func (ds *DataStore) Get(order string, skip, take int, startDate, finishDate tim
 	}).Skip(skip).Sort(order).Limit(take).All(&data)
 	if err != nil {
 		ds.logger.Error(err)
+		return nil, err
 	}
 
 	if data == nil {
 		data = []Data{}
 	}
 
-	return data, err
+	return data, nil
 }
 
 func (ds *DataStore) Save(data Data) error {
 	err := ds.Database.C(collection).Insert(&data)
 	if err != nil {
 		ds.logger.Error(err)
+		return err
 	}
-	return err
+	return nil
+}
+
+func (ds *DataStore) Delete(deviceID string) error {
+	if deviceID != "" {
+		_, err := ds.Database.C(collection).RemoveAll(bson.M{"from": deviceID})
+		if err != nil {
+			ds.logger.Error(err)
+			return err
+		}
+	} else {
+		_, err := ds.Database.C(collection).RemoveAll(nil)
```

Comment:
Couldn't this code be simplified to avoid this second call and those nested conditionals?

Reply:
Something along these lines should work (not sure, tho) :point_down: 

```Go
func (ds *DataStore) Delete(deviceID string) error {
	if deviceID == "" {
		return ds.removeAll(nil)
	}
	return ds.removeAll(bson.M{"from": deviceID})
}

func (ds *DataStore) removeAll(query <type>) error {
	_, err := ds.Database.C(collection).RemoveAll(query)
	if err != nil {
		ds.logger.Error(err)
		return err
	}
	return nil
}
```

Reply:
Good suggestion @lcbm!

Reply:
This conversation is still pending :stuck_out_tongue:

Reply:
@Igoraamc I think the @lcbm suggestion is a good one. :)